x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -290,6 +300,17 @@ void handleReadResponse(long from, long maxRequiredSeqNo, ShardChangesAction.Res
         maybeUpdateSettings(response.getSettingsVersion(), updateMappingsTask);
     }
 
+    void handleFallenBehindLeaderShard(Exception e) {
+        if (fallenBehindLeaderShard.compareAndSet(false, true)) {
```

Comment:
Let's make this method noop for now (i.e., without the state `fallenBehindLeaderShard`). I feel we need a more robust approach to avoid the scenario where an outstanding request can trigger another restore while the shard was restored already.